### PR TITLE
Feature/polkadot0.9.29 dev

### PR DESCRIPTION
**Pull request by baidang201**

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2022-01-01
+nightly-2022-09-01
```
Comment (xcaptain) [inline rust-toolchain:1]: I think it's better to use a `rust-toolchain.toml` here, https://rust-lang.github.io/rustup/overrides.html

image
So no need to manually run `rustup target add wasm32-unknown-unknown`  as the readme says.

File: test-client/Cargo.toml [hunk 2]
```diff
@@ -19,16 +19,16 @@ hash-db = "0.15.2"
...
-sc-service = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1", default-features = false, features = ["test-helpers"]}
-sp-blockchain = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-consensus = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-keystore = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-keyring = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-state-machine = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
+sc-client-api = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.29" }
+sc-client-db = { version = "0.10.0-dev", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.29", features = ["test-helpers"]}
```
Comment (xcaptain) [inline test-client/Cargo.toml:23]: same version thing, why no need to specify version before and you add now.

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -812,6 +816,7 @@ impl pallet_treasury::Config for Runtime {
     type SpendFunds = Bounties;
     type WeightInfo = pallet_treasury::weights::SubstrateWeight<Runtime>;
     type MaxApprovals = MaxApprovals;
+    type SpendOrigin = frame_support::traits::NeverEnsureOrigin<u128>;
```
[the remark this answers is not in the document]
Comment (baidang201) [reply runtime/src/lib.rs:819]: fix

File: test-client/Cargo.toml (hunk 2 above)
Comment (baidang201) [reply test-client/Cargo.toml:23]: fixed

File: inspect/Cargo.toml [hunk 4]
```diff
@@ -14,12 +14,12 @@ targets = ["x86_64-unknown-linux-gnu"]
...
-sc-client-api = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sc-executor = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sc-service = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-blockchain = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
-sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.24-1" }
+sc-cli = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.29" }
+sc-client-api = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.29" }
+sc-executor = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.29" }
+sc-service = { version = "0.10.0-dev", default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.29" }
```
[the remark this answers is not in the document]
Comment (baidang201) [reply inspect/Cargo.toml:20]: fixed

**Review by AmadeusGB — APPROVED** [review]

**Review by xcaptain — APPROVED** [review]

**Pull request merged**